packages/microcosm-examples/canvas/app/boot.js

```diff
@@ -1,13 +1,17 @@
 import Microcosm, { update } from 'microcosm'
+import CompareTree from 'microcosm/addons/compare-tree'
 import prepareCanvas from './prepare-canvas'
 
 let repo = new Microcosm({ batch: true })
+let tree = new CompareTree()
+
+repo.on('change', tree.update, tree)
 
 let size = 20
 let limit = Infinity
 let rows = Math.min((window.innerWidth / size) | 0, limit)
 let columns = Math.min((window.innerHeight / size) | 0, limit)
-let writes = size * 25
+let writes = size * 50
```

Comment:
Something about this change or a prior change caused a big perf improvement. I wish I knew what it was. I'll make a note to dig into this.